src/chainrules.jl

```diff
@@ -0,0 +1,74 @@
+ChainRulesCore.@scalar_rule(airyai(x), airyaiprime(x))
+ChainRulesCore.@scalar_rule(airyaiprime(x), x * airyai(x))
+ChainRulesCore.@scalar_rule(airybi(x), airybiprime(x))
+ChainRulesCore.@scalar_rule(airybiprime(x), x * airybi(x))
+ChainRulesCore.@scalar_rule(besselj0(x), -besselj1(x))
+ChainRulesCore.@scalar_rule(
+    besselj1(x),
+    (besselj0(x) - besselj(2, x)) / 2,
+)
+ChainRulesCore.@scalar_rule(bessely0(x), -bessely1(x))
+ChainRulesCore.@scalar_rule(
+    bessely1(x),
+    (bessely0(x) - bessely(2, x)) / 2,
+)
+ChainRulesCore.@scalar_rule(dawson(x), 1 - (2 * x * Ω))
+ChainRulesCore.@scalar_rule(digamma(x), trigamma(x))
+ChainRulesCore.@scalar_rule(erf(x), (2 / sqrt(π)) * exp(-x * x))
+ChainRulesCore.@scalar_rule(erfc(x), -(2 / sqrt(π)) * exp(-x * x))
+ChainRulesCore.@scalar_rule(erfcinv(x), -(sqrt(π) / 2) * exp(Ω^2))
+ChainRulesCore.@scalar_rule(erfcx(x), (2 * x * Ω) - (2 / sqrt(π)))
+ChainRulesCore.@scalar_rule(erfi(x), (2 / sqrt(π)) * exp(x * x))
+ChainRulesCore.@scalar_rule(erfinv(x), (sqrt(π) / 2) * exp(Ω^2))
+ChainRulesCore.@scalar_rule(gamma(x), Ω * digamma(x))
+ChainRulesCore.@scalar_rule(
+    invdigamma(x),
+    inv(trigamma(invdigamma(x))),
+)
+ChainRulesCore.@scalar_rule(trigamma(x), polygamma(2, x))
+
+# binary
+ChainRulesCore.@scalar_rule(
+    besselj(ν, x),
+    (NaN, (besselj(ν - 1, x) - besselj(ν + 1, x)) / 2),
```

Comment:
See the discussion in https://github.com/JuliaDiff/ChainRules.jl/issues/292 … we should consider switching to`@thunk error("unimplemented")` instead of `NaN` here.